### PR TITLE
Deployment preparation (1/2): Composable stable pool V3

### PR DESCRIPTION
**Pull request by jubeira**
# Deployment preparation: `ComposableStablePool` V3

**Build info: pending**
This is a draft with the groundwork required to do the deployment preparation once the artifact is ready.

Basic fork test should work with composable stable pool V2 artifacts.
To test locally, copy those artifacts from the deprecated task into this one, and remove the skip directive from the fork test.

The fork test shall be enabled once the artifacts are ready, but this can be reviewed and merged in the meantime.

## Deployment preparation tasks checklist:

- [x] Deployments `README` is updated with the new task name and link
- [ ] Task `README` has a description and links to the artifacts and expected output folders 
- [ ] `build-info` is updated with artifacts compiled at the specified commit
- [ ] Artifacts are generated from `build-info`
- [ ] New task has a working fork test 

## Deprecating old tasks checklist: 

- [x] Old task was moved to `deprecated/` directory
- [ ] Deployments `README` is updated: old task is moved to the 'deprecated' section, and links are updated with the `deprecated/` prefix
- [x] Old task imports in `index.ts`, `input.ts` and fork tests are updated
- [ ] Old task `README` is updated with a warning sign, a link to the replacement task and a short description 

## Code checklist:

- [ ] The diff is legible and has no extraneous changes 
- [ ] Complex code has been commented in tests and scripts
- [ ] Fork test block number and test cases are appropriate for this deployment
- [x] The base branch is either `master`, or there's a description of how to merge

## Issue Resolution

N/A

### CODE REVIEW

File: pkg/deployments/README.md [hunk 1]
```diff
@@ -104,6 +104,7 @@ Returns an object with all contracts from a deployment and their addresses.
 | L2 Gauge Checkpointer                                   | [`20221205-l2-gauge-checkpointer`](./tasks/20221205-l2-gauge-checkpointer)                           |
 | VeBoost V2                                              | [`20221205-veboost-v2`](./tasks/20221205-veboost-v2)                                                 |
 | Gauge Registrant V3, supporting the Adaptor entrypoint  | [`20230109-gauge-adder-v3`](./tasks/20230109-gauge-adder-v3)                                         |
+| Composable Stable Pools V3                              | [`20220120-composable-stable-pool-v3`](./tasks/20220130-composable-stable-pool-v3)                   |
```
Comment (EndymionJkb) [inline pkg/deployments/README.md:107]:
```suggestion
| Composable Stable Pools V3                              | [`20220130-composable-stable-pool-v3`](./tasks/20220130-composable-stable-pool-v3)                   |
```

File: pkg/deployments/tasks/20230130-composable-stable-pool-v3/readme.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# 2023-01-23 - Composable Stable Pool V3
```
Comment (EndymionJkb) [inline pkg/deployments/tasks/20230130-composable-stable-pool-v3/readme.md:1]:
```suggestion
# 2023-01-30 - Composable Stable Pool V3
```

File: pkg/deployments/tasks/20230130-composable-stable-pool-v3/test/test.fork.ts [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+import { getSigner, impersonate } from '../../../src/signers';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+
+import { MAX_UINT256, ZERO_ADDRESS } from '@balancer-labs/v2-helpers/src/constants';
+import { bn, fp } from '@balancer-labs/v2-helpers/src/numbers';
+import { BasePoolEncoder, StablePoolEncoder, SwapKind } from '@balancer-labs/balancer-js';
+import { actionId } from '@balancer-labs/v2-helpers/src/models/misc/actions';
+import { expectEqualWithError } from '@balancer-labs/v2-helpers/src/test/relativeError';
+
+describeForkTest.only('ComposableStablePool', 'mainnet', 16492500, function () {
```
Comment (jubeira) [inline pkg/deployments/tasks/20230130-composable-stable-pool-v3/test/test.fork.ts:18]: TODO: remove this.

Comment (jubeira) [inline pkg/deployments/tasks/20230130-composable-stable-pool-v3/test/test.fork.ts:18]:
```suggestion
describeForkTest('ComposableStablePool', 'mainnet', 16492500, function () {
```

Remove `.only` when ready.

File: pkg/deployments/README.md [hunk 4]
```diff
@@ -104,6 +104,7 @@ Returns an object with all contracts from a deployment and their addresses.
 | L2 Gauge Checkpointer                                   | [`20221205-l2-gauge-checkpointer`](./tasks/20221205-l2-gauge-checkpointer)                           |
 | VeBoost V2                                              | [`20221205-veboost-v2`](./tasks/20221205-veboost-v2)                                                 |
 | Gauge Registrant V3, supporting the Adaptor entrypoint  | [`20230109-gauge-adder-v3`](./tasks/20230109-gauge-adder-v3)                                         |
+| Composable Stable Pools V3                              | [`20230130-composable-stable-pool-v3`](./tasks/20230130-composable-stable-pool-v3)                   |
```
Comment (nventuro) [inline pkg/deployments/README.md:107]: Change date to feb 6th

**Comment by jubeira** [comment]
Ready to review at https://github.com/balancer-labs/balancer-v2-monorepo/pull/2190/commits/5e5f585dedccf27f66dcba37226cb13dffa12996.

Basic fork test should work with composable stable pool V2 artifacts.
To test locally, copy those artifacts from the deprecated task into this one, and remove the `skip` directive from the fork test.

The fork test shall be enabled once the artifacts are ready, but this can be reviewed and merged in the meantime.

File: pkg/deployments/README.md [hunk 5]
```diff
@@ -104,7 +104,7 @@ Returns an object with all contracts from a deployment and their addresses.
 | L2 Gauge Checkpointer                                   | [`20221205-l2-gauge-checkpointer`](./tasks/20221205-l2-gauge-checkpointer)                           |
 | VeBoost V2                                              | [`20221205-veboost-v2`](./tasks/20221205-veboost-v2)                                                 |
 | Gauge Registrant V3, supporting the Adaptor entrypoint  | [`20230109-gauge-adder-v3`](./tasks/20230109-gauge-adder-v3)                                         |
-| Composable Stable Pools V3                              | [`20230130-composable-stable-pool-v3`](./tasks/20230130-composable-stable-pool-v3)                   |
+| Composable Stable Pools V3                              | [`20220206-composable-stable-pool-v3`](./tasks/20220206-composable-stable-pool-v3)                   |
```
Comment (nventuro) [inline pkg/deployments/README.md:107]: 2023, not 2022

**Review by nventuro — APPROVED** [review]

**Pull request merged**